src/macros.jl

```diff
@@ -359,23 +368,46 @@ macro wrap(ex)
 
     fullargs = ex.args[2:length(ex.args)]
 
-    # strip type annotation from arguments for use in the call to GAP
-    args = [x isa Symbol ? x : x.args[1] for x in fullargs]
-
+    # splits the arguments with type annotations into expressions for the lhs and rhs
+    # of the call of the form `func(args) = GAP.Globals.func(args)`:
+    # - type annotations have to be removed for the rhs
+    # - if the type annotation is `GapObj`, the rhs has to be wrapped in `GAP.GapObj(...)` and the lhs has no type annotation
```

Comment:
Should we also extend the comment to match the code?
And/or add relevant comments next to the code.```suggestion
    # - if the type annotation is `GapObj`, `GapInt`, or a few other specific cases (see the code for details), the rhs is wrapped in `GAP.GapObj(...)` and we drop the type annotation for the lhs
```
(and perhaps line-wrap this)